### PR TITLE
feat: Add /buyers/billing_details endpoints [TA-1022]

### PR DESCRIPTION
**Pull request by szeying-lee**
# Description
Documents new `PUT /buyers/billing_details` and `GET /buyers/billing_details` endpoints. Also recreates `address_update_request.yaml` as all address fields are actually optional (https://github.com/gr4vy/core-api/blob/main/src/api/routers/buyers/schemas.py#L91).

Fixes # (issue)
https://gr4vy.atlassian.net/browse/TA-1022

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own changes
- [ ] I have run `yarn test` to make sure my changes pass all linters
- [ ] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: reference/openapi.yaml [hunk 1]
```diff
@@ -187,6 +187,12 @@ paths:
     delete:
       $ref: "./paths/buyers__delete-buyer.yaml"
 
+  "/buyers/billing_details":
```
Comment (thomas-gr4vy) [inline reference/openapi.yaml:190]:
```suggestion
  "/buyers/billing-details":
```

File: reference/paths/buyers__get-buyer-billing-details.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  - $ref: "../query-parameters/buyer_id.yaml"
+  - $ref: "../query-parameters/buyer_external_identifier.yaml"
+
+responses:
+  "200":
+    description: Returns the information about a buyer's billing details.
+    content:
+      application/json:
+        schema:
+          $ref: "../openapi.yaml#/components/schemas/BillingDetails"
```
Comment (thomas-gr4vy) [inline reference/paths/buyers__get-buyer-billing-details.yaml:23]: Do we need to extend this to add a `type` of `billing-details` (and what about an `id`)?

**Review by thomas-gr4vy — COMMENTED** [review]
Added a few comments.

File: reference/request-bodies/address_update_request.yaml [hunk 3]
```diff
@@ -5,15 +5,81 @@ type: object
...
-  - line1
-  - line2
-  - city
-  - state
-  - postal_code
-  - country
-
-allOf:
-  - $ref: "../openapi.yaml#/components/schemas/Address"
+properties:
```
Comment (cbetta) [inline reference/request-bodies/address_update_request.yaml:8]: This schema is also used in other requests? Is this changing the behaviour of these? E.g. is it setting fields as not required in them?

Comment (szeying-lee) [reply reference/request-bodies/address_update_request.yaml:8]: Yes, `AddressUpdate` is used in `BillingDetailsUpdateRequest` which is used in `BuyerUpdate`. Have added a link in the description describing this change as none of the fields in `AddressUpdate` is actually required: https://github.com/gr4vy/core-api/blob/main/src/api/routers/buyers/schemas.py#L91

Comment (cbetta) [reply reference/request-bodies/address_update_request.yaml:8]: @szeying-lee perfect.

File: reference/paths/buyers__get-buyer-billing-details.yaml (hunk 2 above)
Comment (cbetta) [reply reference/paths/buyers__get-buyer-billing-details.yaml:23]: Agreed with this. Not sure what the ID would be though.

Comment (cbetta) [reply reference/paths/buyers__get-buyer-billing-details.yaml:23]: FWIW we could add the type to all BillingDetails records, even those nested in buyers. The ID could be a unique UUID. Not sure if we have a UUID already internally?

**Review by thomas-gr4vy — APPROVED** [review]

File: reference/resources/billing_details.yaml [hunk 4]
```diff
@@ -1,18 +1,25 @@
...
   - Buyers
 
 properties:
+  type:
+    type: string
+    description: The type of this resource. Is always `billing-details`.
+    example: billing-details
+    enum:
+      - billing-details
+
```
Comment (cbetta) [inline reference/resources/billing_details.yaml:17]: Does this still need an ID? @thomas-gr4vy ?

Comment (thomas-gr4vy) [reply reference/resources/billing_details.yaml:17]: Thinking about it, I'm not sure it makes sense to add it at this point in time. There is no way to use the ID to access the resource on its own. Maybe one for the future if/when we support multiple billing details per buyer?

**Pull request merged**